### DISCUSSION TITLE
I got a problem using httpx with gin and asynq

### DISCUSSION THREAD

**Discussion by Lance621** [projectdiscovery/httpx#1677] [Q&A]
I build up a WaitGroup while I use the httpx in the gin handler, but there is no return and wg.Done() dose not been called. 
And I tested in a single file with the same code, it works.

- it works when I called the function in a single file
<img width="990" alt="image" src="https://github.com/projectdiscovery/httpx/assets/18532708/1a6c371b-a5ce-4274-b86d-149e11a14786">

- it doesn't work when I called the function in the gin handler 
<img width="820" alt="image" src="https://github.com/projectdiscovery/httpx/assets/18532708/5383a2e4-ec86-451d-b6c2-daf21f634ee5">
<img width="982" alt="image" src="https://github.com/projectdiscovery/httpx/assets/18532708/61b03807-dd50-492c-b2ca-a83775c15af0">

anyone can give me a tip? I am so comfused.

**Comment by dogancanbakir** [comment]
This discussion closed automatically due to inactivity. Feel free to reopen or start new if still relevant.
